### PR TITLE
GAWB-2852: admin endpoint to get user status by email

### PR DESCRIPTION
**Pull request by davidangb**
Admin endpoint to return a user's status from their email. This also allows email->subjectid lookups, which is a prereq for some free tier functionality.

Also includes swagger yaml fixes, based on warnings/errors shown by editor.swagger.io

I have not updated firecloud-admin-cli. It looks like that is far out of date and doesn't know how to connect to sam for anything - so I haven't taken on that job.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala [hunk 1]
```diff
@@ -45,6 +45,15 @@ trait UserRoutes extends UserInfoDirectives {
       requireUserInfo { userInfo =>
         asWorkbenchAdmin(userInfo) {
           pathPrefix("user") {
+            path("email" / Segment) { email =>
+              complete {
+                userService.getUserStatusFromEmail(email).map { statusOption =>
+                  statusOption.map { status =>
+                    StatusCodes.OK -> Option(status)
+                  }.getOrElse(StatusCodes.NotFound -> None)
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala:53]: what does None actually look like serialized to json?

**Review by dvoet — APPROVED** [review]

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala (hunk 1 above)
Comment (davidangb) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala:53]: it's an empty response (and great question, I didn't think about it, copied this from `/api/admin/user/{userId}`)

**Comment by kyuksel** [comment]
@davidangb Do we still need this endpoint since the free trial users now must register in FireCloud before requesting trial credits, which means we shouldn't need to resort to using their email addresses as key in lookups?

**Review by kyuksel — APPROVED** [review]
Verified the endpoint functionality on local Swagger and that the tests pass.

**Pull request merged**